demos/common/cpp/models/src/detection_model_centernet.cpp

```diff
@@ -0,0 +1,307 @@
+    std::string imageInputName = inputInfo.begin()->first;
+    inputsNames.push_back(imageInputName);
+    netInputHeight = getTensorHeight(inputDesc);
+    netInputWidth = getTensorWidth(inputDesc);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    slog::info << "Checking that the outputs are as the demo expects" << slog::endl;
+
+    InferenceEngine::OutputsDataMap outputInfo(cnnNetwork.getOutputsInfo());
+    if (outputInfo.size() != 3) {
+        throw std::logic_error("This demo expect networks that have 3 outputs blobs");
+    }
+
+    const InferenceEngine::TensorDesc& outputDesc = outputInfo.begin()->second->getTensorDesc();
+    for (auto& output : outputInfo) {
+        output.second->setPrecision(InferenceEngine::Precision::FP32);
+        output.second->setLayout(InferenceEngine::Layout::NCHW);
+        outputsNames.push_back(output.first);
+    }
+
+}
+
+cv::Point2f getDir(const cv::Point2f& srcPoint, float rotRadius) {
+    float sn = sinf(rotRadius);
+    float cs = cosf(rotRadius);
+
+    cv::Point2f srcResult(0.0f, 0.0f);
+    srcResult.x = srcPoint.x * cs - srcPoint.y * sn;
+    srcResult.y = srcPoint.x * sn + srcPoint.y * cs;
+
+    return srcResult;
+}
+
+cv::Point2f get3rdPoint(const cv::Point2f& a, const cv::Point2f& b) {
+    cv::Point2f direct = a - b;
+    return b + cv::Point2f(-direct.y, direct.x);
+}
+
+cv::Mat getAffineTransform(float centerX, float centerY, int scale, float rot, size_t outputWidth, size_t outputHeight, bool inv = false) {
+    int srcW = scale;
+    float rotRad =  static_cast<float>(CV_PI) * rot / 180.0f;
+    auto srcDir = getDir({ 0.0f, -0.5f * srcW }, rotRad);
+    cv::Point2f dstDir(0.0f,  -0.5f * outputWidth);
+    std::vector<cv::Point2f> src(3, { 0.0f, 0.0f });
+    std::vector<cv::Point2f> dst(3, { 0.0f, 0.0f });
+
+    src[0] = { centerX, centerY };
+    src[1] = srcDir + src[0];
+    src[2] = get3rdPoint(src[0], src[1]);
+
+    dst[0] = { outputWidth * 0.5f, outputHeight * 0.5f };
+    dst[1] = dst[0] + dstDir;
+    dst[2] = get3rdPoint(dst[0], dst[1]);
+
+    cv::Mat trans;
+    if (inv) {
+        trans = cv::getAffineTransform(dst, src);
+    }
+    else {
+        trans = cv::getAffineTransform(src, dst);
+    }
+
+    return trans;
+}
+
+std::shared_ptr<InternalModelData> ModelCenterNet::preprocess(const InputData& inputData, InferenceEngine::InferRequest::Ptr& request) {
+    auto& img = inputData.asRef<ImageInputData>().inputImage;
+
+    int imgWidth = img.cols;
+    int imgHeight = img.rows;
+    float centerX = imgWidth / 2.0f;
+    float centerY = imgHeight / 2.0f;
+    int scale = std::max(imgWidth, imgHeight);
+
+    auto transInput = getAffineTransform(centerX, centerY, scale, 0, netInputWidth, netInputHeight);
+    cv::Mat resizedImg;
+    cv::warpAffine(img, resizedImg, transInput, cv::Size(netInputWidth, netInputHeight), cv::INTER_LINEAR);
+    cv::Mat chwImg;
+
+    if (useAutoResize) {
```

Comment:
The size is already set in `cv::warpAffine()`. You should just go with `wrapMat2Blob()` and `NHWC` layout. `ModelCenterNet` shouldn't take `useAutoResize` arg. Set this field to false unconditionally.

Reply:
done